### PR TITLE
move Sunrise to Play 2.5.0, close #198

### LINKED ISSUE #198: Migrate to Play 2.5
Play 2.5-M1

### PR DESCRIPTION
**Pull request by schleichardt**
@lauraluiz please review

### CODE REVIEW

File: shopping-cart/app/shoppingcart/common/CartController.java [hunk 1]
```diff
@@ -60,24 +62,24 @@ public CartController(final ControllerDependency controllerDependency) {
...
+                .thenComposeAsync(cart -> validCartOrNew(cart, userContext), HttpExecution.defaultContext());
     }
 
-    private F.Promise<Cart> fetchCartByCustomerOrNew(final String customerId, final UserContext userContext) {
+    private CompletionStage<Cart> fetchCartByCustomerOrNew(final String customerId, final UserContext userContext) {
         final CartByCustomerIdGet query = CartByCustomerIdGet.of(customerId)
                 .withExpansionPaths(m -> m.shippingInfo().shippingMethod());
         return sphere().execute(query)
-                .flatMap(cart -> validCartOrNew(cart, userContext));
+                .thenComposeAsync(cart -> validCartOrNew(cart, userContext), HttpExecution.defaultContext());
```
Comment (lauraluiz) [inline shopping-cart/app/shoppingcart/common/CartController.java]: It became quite ugly now :(
I'll have to find a solution that doesn't pollute the code so much.

File: pt/controllers/StatusControllerTest.java [hunk 2]
```diff
@@ -50,22 +50,22 @@ public void showsVersion() throws Exception {
...
         assertThat(result.status()).isEqualTo(Http.Status.OK);
-        assertThat(result.contentType()).isEqualTo(Http.MimeTypes.JSON);
+        assertThat(result.contentType()).contains(Http.MimeTypes.JSON);
         assertThat(contentAsString(result)).contains("\"healthy\" : true");
     }
 
-    private void rendersUnhealthyCtp(final StatusController controller) throws IOException {
-        final Result result = controller.health().get(ALLOWED_TIMEOUT);
+    private void rendersUnhealthyCtp(final StatusController controller) throws Exception {
+        final Result result = controller.health().toCompletableFuture().get(1, TimeUnit.SECONDS);
```
Comment (lauraluiz) [inline pt/controllers/StatusControllerTest.java]: Why not join?

File: product-catalog/test/productcatalog/models/BreadcrumbDataTest.java [hunk 3]
```diff
@@ -72,7 +72,7 @@ private void testCategoryBreadcrumb(final String extId, final CategoryTree categ
     }
 
     private void testProductBreadcrumb(final String sku, final ProductProjection product, final Consumer<List<String>> texts, final Consumer<List<String>> urls) {
-        final ProductVariant variant = product.findVariantBySky(sku).get();
+        final ProductVariant variant = product.findVariantBySku(sku).get();
```
Comment (lauraluiz) [inline product-catalog/test/productcatalog/models/BreadcrumbDataTest.java]: I also fixed that in my working branch :P

File: pt/controllers/StatusControllerTest.java (hunk 2 above)
Comment (schleichardt) [inline pt/controllers/StatusControllerTest.java]: join, has no checked exceptions, but here with timeout makes sure the app does not hang

File: shopping-cart/app/shoppingcart/common/CartController.java (hunk 1 above)
Comment (schleichardt) [inline shopping-cart/app/shoppingcart/common/CartController.java]: on some code locations you can get rid of ```, HttpExecution.defaultContext()```, but then you can't use controller methods like ```request()```

Comment (lauraluiz) [inline shopping-cart/app/shoppingcart/common/CartController.java]: I don't want to get rid of it, I just want to think if there's a way to hide this complexity. Before you could easily read and understand the code, but now this distracts a lot.

**Comment by lauraluiz** [comment]
Ta-da!

**Pull request merged**